src/_P144_Vindriktning.ino

```diff
@@ -263,117 +227,14 @@ boolean Plugin_144(uint8_t function, struct EventStruct *event, String& string)
     {
       // code to be executed 10 times per second. Tasks which require fast response can be added here
       // be careful on what is added here. Heavy processing will result in slowing the module down!
-      bool new_data = false;
-      for (int charAvailable = P144_easySerial->available(); ((charAvailable > 0) && !new_data); charAvailable--)
-      {
-        // Process received characters, return true when a complete message is received 
-        // Message rate on Vindriktning is a few per minute
-        new_data = P144_process_rx(P144_easySerial->read());
-        if (new_data) 
-        {
-          P144_pm25 = float((P144_serialRxBuffer[3] << 8) + P144_serialRxBuffer[4]);
-          #ifdef PLUGIN_144_DEBUG
-          if (loglevelActiveFor(LOG_LEVEL_INFO))
-          {
-            String log = F("P144 : New value received ");
-            log += P144_pm25;
-            addLogMove(LOG_LEVEL_INFO, log);
-          }
-          #endif
-        }
-      }
-      success = true;
+      P144_data_struct *P144_data =
+        static_cast<P144_data_struct *>(getPluginTaskData(event->TaskIndex));
+      success = P144_data->processSensor();
```

Comment:
You need to check for nullptr before using `->`

Reply:
Thanks, fixed